### DISCUSSION TITLE
How to deploy across multiple datacenters?

### DISCUSSION THREAD

**Discussion by wushengwusheng** [ccfos/nightingale#1238] [Q&A] [answered]
目前业务有部署多个机房分别在不同的城市，中间可以专线互通。如果想采取共一个前端页面，后端各自部署的架构，应该怎么部署？求指教

**Comment by FengZh61** [comment, marked as answer]
B站看 BV1vb4y1i7a9 这个视频 讲的很详细。

**Reply by wushengwusheng** [reply]
感谢回复。我就是这么配置，在server.conf里配置了不同的Name，webapi.conf里配置了多个clusters,n9e dashboard右上角下拉框也有了选项了，但是集群上只读到了webapi.conf第一个[[clusters]],版本为5.13，不确定是不是bug

**Reply by wushengwusheng** [reply]
<img width="704" alt="image" src="https://user-images.githubusercontent.com/30250875/199181664-a222d55b-ada3-4638-8c32-d12ea85ff6f2.png">
<img width="1004" alt="image" src="https://user-images.githubusercontent.com/30250875/199181737-3dde3454-ee1f-4944-ae77-157072275bbc.png">
![image](https://user-images.githubusercontent.com/30250875/199181882-9d40f90d-1acb-4148-b862-7e3d06de3215.png)
![image](https://user-images.githubusercontent.com/30250875/199181940-1eda78b4-0ee0-4bcd-8224-8c03b545418c.png)
三台机器server.conf 分别是hz,wx,bj,webapi.conf中三个[[clusters]] Name分别为wx，hz,bj，但是显示出来的机器集群全是hz

**Reply by FengZh61** [reply]
> <img alt="image" width="704" src="https://user-images.githubusercontent.com/30250875/199181664-a222d55b-ada3-4638-8c32-d12ea85ff6f2.png">
> 
> <img alt="image" width="1004" src="https://user-images.githubusercontent.com/30250875/199181737-3dde3454-ee1f-4944-ae77-157072275bbc.png">
> 
> ![image](https://user-images.githubusercontent.com/30250875/199181882-9d40f90d-1acb-4148-b862-7e3d06de3215.png) ![image](https://user-images.githubusercontent.com/30250875/199181940-1eda78b4-0ee0-4bcd-8224-8c03b545418c.png) 三台机器server.conf 分别是hz,wx,bj,webapi.conf中三个[[clusters]] Name分别为wx，hz,bj，但是显示出来的机器集群全是hz

我没看明白有啥问题啊

**Reply by wushengwusheng** [reply]
按我这么配置 192.168.12.134集群应该是wx
<img width="601" alt="image" src="https://user-images.githubusercontent.com/30250875/199189004-193a582a-b1b5-4f67-91f1-af0008f13820.png">
<img width="311" alt="image" src="https://user-images.githubusercontent.com/30250875/199189057-f7f72f5c-4c40-486f-9ae8-30a50e43e5e8.png">

**Comment by UlricQin** [comment]
除了视频，文档里也有一些解释：http://n9e.flashcat.cloud/docs/install/server/

**Reply by wushengwusheng** [reply]
我看到了 就是照着配置的～～但是三台集群集群都是同一个 不知道是哪里配错了

**Reply by FengZh61** [reply]
> 我看到了 就是照着配置的～～但是三台集群集群都是同一个 不知道是哪里配错了

先在prometheus确认下数据是否正常，然后查一下n9e的配置。

**Reply by wushengwusheng** [reply]
请问您那边多集群配置是正常的 版本是多少～～

**Reply by FengZh61** [reply]
> 请问您那边多集群配置是正常的 版本是多少～～
我用的最新版

**Reply by wushengwusheng** [reply]
我版本5.13，我试了下5.11也不行 方便把server.conf和webapi.conf贴一下吗

**Reply by FengZh61** [reply]
> 我版本5.13，我试了下5.11也不行 方便把server.conf和webapi.conf贴一下吗

没啥特殊的，server的配置对应每个prometheus节点，webapi的配置对应每个prometheus节点。

**Reply by wushengwusheng** [reply]
那我再试试吧

**Reply by wushengwusheng** [reply]
@FengZh61 可以截图看下吗 n9e dashboard 监控对象->对象列表

**Reply by FengZh61** [reply]
> @FengZh61 可以截图看下吗 n9e dashboard 监控对象->对象列表

你用的Prometheus数据源，对象列表怎么会有数据呢？如果你的数据没有经过n9e server，是不会生成监控对象的，你可以看看你的数据库，有几条数据。

**Reply by wushengwusheng** [reply]
![image](https://user-images.githubusercontent.com/30250875/199407050-beaaff35-3d48-46f9-a180-48f3b0c3e4c4.png)
![image](https://user-images.githubusercontent.com/30250875/199407073-54caec69-bc06-4f3b-8170-975354151b1b.png)
server.conf里读写都是往prometheus里的，我的对象列表是有数据的
<img width="1403" alt="image" src="https://user-images.githubusercontent.com/30250875/199407239-cb3252cb-7bc4-45da-86e4-d6c8831de542.png">

**Reply by FengZh61** [reply]
> ![image](https://user-images.githubusercontent.com/30250875/199407050-beaaff35-3d48-46f9-a180-48f3b0c3e4c4.png) ![image](https://user-images.githubusercontent.com/30250875/199407073-54caec69-bc06-4f3b-8170-975354151b1b.png) server.conf里读写都是往prometheus里的，我的对象列表是有数据的 <img alt="image" width="1403" src="https://user-images.githubusercontent.com/30250875/199407239-cb3252cb-7bc4-45da-86e4-d6c8831de542.png">

我理解你的数据流是采集器到n9e server 到 prometheus里。采集器数据经过n9e server后，写入Prometheus。我用的路子跟你不一样，我是直接用n9e对接的Prometheus数据源，我的数据没有经过n9e server。如果你想你的监控节点出现到监控对象里面，那么你的label中一定要有ident=“xxx”，你看看你没上线的那几个数据源里面，有没有这个label。

**Reply by wushengwusheng** [reply]
我想确定下 你对象列表里的集群可以是多个吗？如果可以的话，你的n9e版本是多少？

**Reply by FengZh61** [reply]
> 我想确定下 你对象列表里的集群可以是多个吗？如果可以的话，你的n9e版本是多少？

我的数据没有经过n9e 所以我的对象列表是空的。跟你的使用方法不一样。我一直以为是你的配置了多集群，在前端差不到另外2个集群的数据呢。

**Reply by wushengwusheng** [reply]
是的。我感觉我们描述的不是一个问题，哈哈哈。我现在的问题是多集群服务器上报到对象列表里，但是各个集群名称都是一样的，没达到多集群的效果。

**Reply by FengZh61** [reply]
> 是的。我感觉我们描述的不是一个问题，哈哈哈。我现在的问题是多集群服务器上报到对象列表里，但是各个集群名称都是一样的，没达到多集群的效果。

给个排查建议吧，你可以把数据库里面的target这个表里面的数据清空，然后等他重新生成监控对象，再看看情况。

**Reply by wushengwusheng** [reply]
![image](https://user-images.githubusercontent.com/30250875/199418189-a8337169-089d-4776-92e2-9f6617f2527e.png)
试过了，重新上报，集群这块都是一样的

**Reply by FengZh61** [reply]
> ![image](https://user-images.githubusercontent.com/30250875/199418189-a8337169-089d-4776-92e2-9f6617f2527e.png) 试过了，重新上报，集群这块都是一样的

你用的什么采集器，分别在3个数据源里查下数据吧，看看是不是都上报到一个n9e server了。假如代码没问题的话，我能想到的只有这一种可能了。。。

**Reply by wushengwusheng** [reply]
是的 我telegraf都往同一个server写了～～光顾着区分数据源了，上报没区分，希望官方文档可以写的明确一些

**Reply by FengZh61** [reply]
> 是的 我telegraf都往同一个server写了～～光顾着区分数据源了，上报没区分，希望官方文档可以写的明确一些

- -解决了就好。

**Reply by wushengwusheng** [reply]
十分感谢老哥的细心解答，再次万分感谢

**Reply by wushengwusheng** [reply]
@FengZh61 再请教个问题 多机房的话各自往各自的n9e server写，假设一个机房500台服务器，需要多少个n9e server呀？或者说如何保证多服务器往n9e server写高可用？

**Reply by FengZh61** [reply]
> @FengZh61 再请教个问题 多机房的话各自往各自的n9e server写，假设一个机房500台服务器，需要多少个n9e server呀？或者说如何保证多服务器往n9e server写高可用？

这个太专业了，需要作者来解答了

**Comment by wushengwusheng** [comment]
@UlricQin 大佬 5.13版本有配置成功的案例吗？ 我看了视频，检查了配置，三台机器三个prometheus，server.conf中clusterName分别为hz，bj，wx，webapi.conf中添加了三个[[clusters]],但是集群显示的都是第一个[[clusters]]

**Reply by UlricQin** [reply]
> @UlricQin 大佬 5.13版本有配置成功的案例吗？ 我看了视频，检查了配置，三台机器三个prometheus，server.conf中clusterName分别为hz，bj，wx，webapi.conf中添加了三个[[clusters]],但是集群显示的都是第一个[[clusters]]

可以微信联系我远程排查，远程排查需要花费较多时间，每次收费78元（解决不了退费）；或采购我司技术支持服务：https://market.aliyun.com/products/53366009/cmfw00060579.html 我们提供原厂7X24小时技术support。

**Reply by wushengwusheng** [reply]
![image](https://user-images.githubusercontent.com/30250875/199417784-1510413f-ea62-402d-a92a-ec6e45e8cfb3.png)
这里的集群永远都是一个 是代码问题吗？配置检查过好多遍了
